core/src/main/java/org/elasticsearch/index/mapper/core/TypeParsers.java

```diff
@@ -284,6 +286,15 @@ public static void parseField(FieldMapper.Builder builder, String name, Map<Stri
                 builder.fieldDataSettings(settings);
                 iterator.remove();
             } else if (propName.equals("copy_to")) {
+                if (parserContext.isWithinMultiField()) {
+                    if (indexVersionCreated.after(Version.V_2_1_0) ||
+                        (indexVersionCreated.after(Version.V_2_0_1) && indexVersionCreated.before(Version.V_2_1_0))) {
+                        throw new MapperParsingException("copy_to in multi fields is not allowed. Found the copy_to in field [" + name + "] which is within a multi field.");
+                    } else {
+                        ESLoggerFactory.getLogger("mapping [" + parserContext.type() + "]").warn("Found a copy_to in field [" + name + "] which is within a multi field. This feature has been removed and the copy_to will be ignored.");
+                        // we still parse this, otherwise the message will only appear once and the copy_to removed. After that it will appear again. Better to have it always.
+                    }
+                }
                 parseCopyFields(propNode, builder);
```

Comment:
Hmm thinking about this again... If you're doing a rolling upgrade, then when the shard is assigned to a new node (as part of the upgrade) it should update the mapping and send the new mapping to the  master. So we should only get this warning once (I think for the first shard?).